src/chttpd/src/chttpd_db.erl

```diff
@@ -254,6 +255,36 @@ handle_view_cleanup_req(Req, Db) ->
     ok = fabric:cleanup_index_files_all_nodes(Db),
     send_json(Req, 202, {[{ok, true}]}).
 
+
+handle_partition_req(#httpd{method='GET', path_parts=[_,_,PartId]}=Req, Db) ->
+    couch_partition:validate_partition(PartId),
+    case couch_db:is_partitioned(Db) of
+        true ->
+            {ok, PartitionInfo} = fabric:get_partition_info(Db, PartId),
+            send_json(Req, {PartitionInfo});
+        false ->
+            throw({bad_request, <<"database is not partitioned">>})
+    end;
+
+handle_partition_req(#httpd{path_parts = [_, _, _]}=Req, _Db) ->
+    send_method_not_allowed(Req, "GET");
+
+handle_partition_req(#httpd{path_parts=[DbName, _, PartId | Rest]}=Req, Db) ->
+    case couch_db:is_partitioned(Db) of
+        true ->
+            QS = chttpd:qs(Req),
+            NewQS = lists:ukeysort(1, [{"partition", ?b2l(PartId)} | QS]),
+            NewReq = Req#httpd{
+                path_parts = [DbName | Rest],
+                qs = NewQS
+            },
+            Handler = chttpd_handlers:db_handler(hd(Rest), fun db_req/2),
+            do_db_req(NewReq, Handler);
+        false ->
+            throw({bad_request, <<"database is not partitioned">>})
+    end.
+
```

Comment:
We probably want to add a base clause here, like:
```erlang
handle_partition_req(Req, Db) ->
    db_req(Req, Db).
```

That'd give a simple 404 on `/#{db}/_partition` instead of the dreadful `function_clause`, `unknown_error` we have there now.

Reply:
I just added a 404 handler directly rather than falling through to db_req which might do something weird.